### PR TITLE
feat(GUI-786): demo the embedded Customer Portal 2.0

### PR DESCRIPTION
**Pull request by rodrigocipriani**
## Why

The Embedded Portal 2.0 flow has shipped (`/embed/customer-login/sso` exchange, dedicated embed session, frame headers), but this repo still demoed the 1.0 flow first and 2.0 as an add-on. Since the demo doubles as provider-facing example code, it should show exactly what a provider integrating the 2.0 portal needs: mint a token server-side, frame the SSO exchange, land in the portal - and nothing else.

## Verification

- `npm run build` and `npm run lint` pass clean.
- Verified end to end against a non-production environment: both versions mint real tokens through `/api/embed-token`, the 2.0 SSO exchange returns 307 to `/embed/portal/{projectId}/{page}` with the embed session cookies set, and the portal renders and navigates inside the iframe when the demo is served over HTTPS same-site with the app.

## Notes

- `NEXT_PUBLIC_DEFAULT_VIEW` is gone: 2.0 is simply the default, and the presence of legacy credentials is what enables the toggle.
- Dev-only: if you serve the dev server through a reverse proxy, the proxy must forward websocket upgrades on `/_next/hmr` (Turbopack's HMR path) or the console fills with socket retries.

## Screenshot

image

## Note to the Reviewer

Review this change together with the authoring developer. While discussing the change, pay special attention to:

- Change Size (< 250 LOC)
- SOLID Principles
- Change Complexity
- User Story Acceptance Criteria
- Automated Tests: automated tests and risk of regressions or uncovered edge cases
- Feature Flags: appropriate code versioning and feature flag placement
- Quality and Performance Code Smells

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1 @@
+@AGENTS.md
```
Comment (RikieriSartor) [inline CLAUDE.md:1]: Do we want to keep the Agents and Claude md files here since it's just demo?

**Review by RikieriSartor — COMMENTED** [review]
Just added a simple comment about the agents md files. Nice job!

File: CLAUDE.md (hunk 1 above)
Comment (rodrigocipriani) [reply CLAUDE.md:1]: These are auto-generated by Next 16: `next dev` writes them on every run (see `node_modules/next/dist/server/lib/generate-agent-files.js`), so deleting them just leaves a permanently dirty working tree. They tell AI coding tools to read the bundled Next 16 docs instead of trusting outdated training data, which is useful in a repo that doubles as example code. Committing them is the intended behavior.

**Review by RikieriSartor — APPROVED** [review]
